docs/DeveloperGuide.md

```diff
@@ -1,33 +1,230 @@
-# Developer Guide
+# ActiveEdge Developer Guide 🥦🏃‍♂️
 
-## Acknowledgements
+## Table of Contents
+- [Introduction](#introduction)
+    - [Purpose](#purpose)
+    - [How to use the Developer Guide](#how-to-use-the-developer-guide)
+- [Quick Start](#quick-start)
+- [Design & Implementation](#design--implementation)
+    - [System Architecture](#system-architecture)
+    - [Main Component](#main-component)
+    - [UI Package](#ui-package)
+    - [Parser Package](#parser-package)
+    - [Command Package](#command-package)
+    - [LogList Package](#entrylist-package)
+    - [Storage Package](#storage-package)
+- [Product Scope](#product-scope)
+    - [Target User Profile](#target-user-profile)
+    - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Manual Testing](#instructions-for-manual-testing)
+- [Command Summary](#command-summary)
+- [Glossary](#glossary)
+- [Acknowledgements](#acknowledgements)
 
```

Comment:
Contents section is good but some links such as "How to use Developer Guide" and "Design Implementation" do not work.